### ISSUE TITLE
[BUG] Subqueries don't seem to work with UNION clauses

### ISSUE THREAD

**Issue by Josipmrden** [memgraph/memgraph#1037]
**Memgraph version**
v2.8

**Environment**
Linux, binary

**Describe the bug**
```
match (n:Node)
match (s:SuperNode {id: 1})
call { with n, s merge (n)<-[:HAS_REL_TO]-(s) }
return s
union all
match (n:Node)
match (s:SuperNode {id: 1})
call { with n, s merge (n)<-[:HAS_REL_TO]-(s) }
return s;
```
Exception: `All subqueries in an UNION must have the same column names.`

**Expected behavior**
It should work as expected

**Comment by gitbuda** [comment]
@Josipmrden can you verify if that is fixed in v2.9

**Comment by gitbuda** [comment]
This seems to be fixed in v2.9
